sycl/test/esimd/sycl-post-link-test.ll

```diff
@@ -0,0 +1,153 @@
+; RUN: sycl-post-link -split-esimd -lower-esimd -O2 -S %s -o %t.table
+; RUN: FileCheck %s -input-file=%t_esimd_0.ll
+; This test checks that IR code below can be successfully processed by
+; sycl-post-link. In this IR no extractelement instruction and no casting are used
+
+; ModuleID = 'sycl-post-link-test.cpp'
+source_filename = "sycl-post-link-test.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%"class.sycl::_V1::ext::intel::esimd::simd" = type { %"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl" }
+%"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl" = type { <8 x float> }
+%"class.sycl::_V1::ext::intel::esimd::simd.0" = type { %"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl.1" }
+%"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl.1" = type { <8 x i32> }
+
+@__spirv_BuiltInGlobalInvocationId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: convergent norecurse
+define dso_local spir_func void @vmult2(%"class.sycl::_V1::ext::intel::esimd::simd"* noundef %a) local_unnamed_addr #0 !srcloc !47 !sycl_explicit_simd !45 !intel_reqd_sub_group_size !48 !sycl_fixed_targets !45 {
+entry:
+  %Res.i.i.i = alloca %"class.sycl::_V1::ext::intel::esimd::simd", align 32
+  %Res.i.i = alloca %"class.sycl::_V1::ext::intel::esimd::simd", align 32
+  %ref.tmp.i = alloca %"class.sycl::_V1::ext::intel::esimd::simd.0", align 32
+  %0 = load i64, i64 addrspace(1)* getelementptr (<3 x i64>, <3 x i64> addrspace(1)* @__spirv_BuiltInGlobalInvocationId, i64 0, i64 0), align 32
+  %conv = trunc i64 %0 to i32
+  %1 = bitcast %"class.sycl::_V1::ext::intel::esimd::simd.0"* %ref.tmp.i to i8*
```

Comment:
1) Please simplify the test. Only instructions participating in pattern matching should be left.
2) Please inline //CHECK: directives right after source IR instruction they are supposed to verify 
transformation of